crates/red_knot_python_semantic/src/types.rs

```diff
@@ -629,6 +629,11 @@ impl<'db> Type<'db> {
                 Type::StringLiteral(_) | Type::LiteralString,
                 Type::Instance(InstanceType { class }),
             ) if class.is_known(db, KnownClass::Str) => true,
+            (
+                Type::Instance(InstanceType { class }),
+                Type::StringLiteral(_) | Type::LiteralString,
+            ) if class.is_known(db, KnownClass::Str) => false,
+            (Type::LiteralString, Type::StringLiteral(_)) => false,
```

Comment:
All of the tests in this PR pass without adding these lines. The default fallback case for `is_subtype_of` is "not a subtype", so generally we don't need to add specific `false` cases like this, unless they need to be handled before a subsequent "true" case.
```suggestion
```